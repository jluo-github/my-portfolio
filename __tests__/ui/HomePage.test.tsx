import { render, screen } from "@testing-library/react";

import HomePage from "@/app/page";

// Mock child components
jest.mock("@/components/Hero", () => {
  const Hero = () => <div data-testid='hero'>Hero Component</div>;
  Hero.displayName = "Hero";
  return Hero;
});
jest.mock("@/components/Experience", () => () => (
  <div data-testid='experience'>Experience Component</div>
));
jest.mock("@/components/About", () => () => (
  <div data-testid='about'>About Component</div>
));
jest.mock("@/components/Projects", () => () => (
  <div data-testid='projects'>Projects Component</div>
));
jest.mock("@/components/Contact", () => () => (
  <div data-testid='contact'>Contact Component</div>
));

describe("Test HomePage", () => {
  test("render HomePage", () => {
    render(<HomePage />);

    expect(screen.getByTestId("hero")).toBeInTheDocument();
    expect(screen.getByTestId("experience")).toBeInTheDocument();
    expect(screen.getByTestId("about")).toBeInTheDocument();
    expect(screen.getByTestId("projects")).toBeInTheDocument();
    expect(screen.getByTestId("contact")).toBeInTheDocument();
  });

  test("matches the snapshot", () => {
    const { asFragment } = render(<HomePage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
