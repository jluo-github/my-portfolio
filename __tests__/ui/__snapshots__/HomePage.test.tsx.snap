// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HomePage matches the snapshot 1`] = `
<DocumentFragment>
  <main
    class="mx-auto sm:px-6 md:px-8 gap-8 sm:gap-12 md:gap-20 flex flex-col w-full"
  >
    <div
      class="w-full h-[2px]"
    />
    <div
      data-testid="hero"
    >
      Hero Component
    </div>
    <div
      data-testid="experience"
    >
      Experience Component
    </div>
    <div
      data-testid="about"
    >
      About Component
    </div>
    <div
      data-testid="projects"
    >
      Projects Component
    </div>
    <div
      data-testid="contact"
    >
      Contact Component
    </div>
  </main>
</DocumentFragment>
`;
